amplify/backend/function/fetchMotionState/src/index.js

```diff
@@ -3,37 +3,91 @@ const {
   getLatestMotionState,
   updateStakerRewardsInDB,
   getMotionData,
+  didMotionPass,
+  updateMotionMessagesInDB,
 } = require('./utils');
-
 /**
  * @type {import('@types/aws-lambda').APIGatewayProxyHandler}
  */
 exports.handler = async (event) => {
   const { colonyAddress, transactionHash } = event.arguments?.input || {};
   /* Get latest motion state from chain */
   const motionData = await getMotionData(transactionHash);
-  const motionState = await getLatestMotionState(colonyAddress, motionData);
+  const motionState =
+    motionData && (await getLatestMotionState(colonyAddress, motionData));
```

Comment:
With #505, you shouldn't need this guard as we shouldn't be calling this function with a tx hash that isn't a motion. In fact, motion data being undefined (and thus triggering a lamda error) is a clue this is being called incorrectly, and thus might be useful.

I'd say if you want to be cautious, wrap everything from line 16 down in:

``` js

if(motionData) {
} 
```

And we can also update the logging logic in `getMotionData` to log if `motionData` is undefined, instead of actionData, so we'll see in the console if motion data isn't found.